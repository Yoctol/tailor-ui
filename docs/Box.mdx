---
name: Box
menu: Layout
route: /layout/box
---

import { PropsTable, Playground, Link } from 'docz';
import { Box } from 'tailor-ui';

# Box

<Link to="/layout/grid#examples">Examples of usage</Link>

## API

All of the properties of the `<Box>` component are **not** required. Pick whatever you want.

<div>
  You can use{' '}
  <Link to="/layout/space#api">the properties of the Space component</Link> to
  modify the elementâ€™s margin or padding.
</div>

You can use these properties as well:

| Property     | Description                      |
| ------------ | -------------------------------- |
| position     | The _position_ CSS property      |
| top          | The _top_ CSS property           |
| right        | The _right_ CSS property         |
| bottom       | The _bottom_ CSS property        |
| left         | The _left_ CSS property          |
| zIndex       | The _z-index_ CSS property       |
| display      | The _display_ CSS property       |
| lineHeight   | The _line-height_ CSS property   |
| background   | The _background_ CSS property    |
| fontSize     | The _font-size_ CSS property     |
| textAlign    | The _text-align_ CSS property    |
| color        | The _color_ CSS property         |
| borderTop    | The _border-top_ CSS property    |
| borderRight  | The _border-right_ CSS property  |
| borderBottom | The _border-bottom_ CSS property |
| borderLeft   | The _border-left_ CSS property   |
| borderColor  | The _border-clolr_ CSS property  |
| borderRadius | The _border-radius_ CSS property |
| width        | The _width_ CSS property         |
| minWidth     | The _min-width_ CSS property     |
| maxWidth     | The _max-width_ CSS property     |
| height       | The _height_ CSS property        |
| minHeight    | The _min-height_ CSS property    |
| maxHeight    | The _max-height_ CSS property    |
| flex         | The _flex_ CSS property          |
| order        | The _order_ CSS property         |
| alignSelf    | The _align-self_ CSS property    |
| overflow     | The _overflow_ CSS property      |
| overflowX    | The _overflow-x_ CSS property    |
| overflowY    | The _overflow-y_ CSS property    |
