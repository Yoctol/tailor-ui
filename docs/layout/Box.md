---
id: box
title: Box
---

[Examples of usage](width-scale.md)

## API

All of the properties of the `<Box>` component are **not** required. Pick whatever you want.

You can use these properties as well:

| Property     | Description                                                   |
| ------------ | ------------------------------------------------------------- |
| position     | The _position_ CSS property                                   |
| top          | The _top_ CSS property                                        |
| right        | The _right_ CSS property                                      |
| bottom       | The _bottom_ CSS property                                     |
| left         | The _left_ CSS property                                       |
| zIndex       | The _z-index_ CSS property                                    |
| display      | The _display_ CSS property                                    |
| lineHeight   | The _line-height_ CSS property                                |
| background   | The _background_ CSS property                                 |
| fontSize     | The _font-size_ CSS property                                  |
| textAlign    | The _text-align_ CSS property                                 |
| color        | The _color_ CSS property                                      |
| borderTop    | The _border-top_ CSS property                                 |
| borderRight  | The _border-right_ CSS property                               |
| borderBottom | The _border-bottom_ CSS property                              |
| borderLeft   | The _border-left_ CSS property                                |
| borderColor  | The _border-clolr_ CSS property                               |
| borderRadius | The _border-radius_ CSS property                              |
| width        | The _width_ CSS property                                      |
| minWidth     | The _min-width_ CSS property                                  |
| maxWidth     | The _max-width_ CSS property                                  |
| height       | The _height_ CSS property                                     |
| minHeight    | The _min-height_ CSS property                                 |
| maxHeight    | The _max-height_ CSS property                                 |
| flex         | The _flex_ CSS property                                       |
| order        | The _order_ CSS property                                      |
| alignSelf    | The _align-self_ CSS property                                 |
| overflow     | The _overflow_ CSS property                                   |
| overflowX    | The _overflow-x_ CSS property                                 |
| overflowY    | The _overflow-y_ CSS property                                 |
| m            | the _margin_ CSS property                                     |
| mt           | the _margin-top_ CSS property                                 |
| mr           | the _margin-right_ CSS property                               |
| mb           | the _margin-bottom_ CSS property                              |
| ml           | the _margin-left_ CSS property                                |
| mx           | the property that set both _margin-left_ and _margin-right_   |
| my           | the property that set both _margin-top_ and _margin-bottom_   |
| p            | the _padding_ CSS property                                    |
| pt           | the _padding-top_ CSS property                                |
| pr           | the _padding-right_ CSS property                              |
| pb           | the _padding-bottom_ CSS property                             |
| pl           | the _padding-left_ CSS property                               |
| px           | the property that set both _padding-left_ and _padding-right_ |
| py           | the property that set both _padding-top_ and _padding-bottom_ |
