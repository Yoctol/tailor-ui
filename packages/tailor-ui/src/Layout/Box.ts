import styled from 'styled-components';
import * as system from 'styled-system';

export type BoxProps = system.GridAreaProps &
  system.PositionProps &
  system.SpaceProps &
  system.BackgroundProps &
  system.TypographyProps &
  system.ColorProps &
  system.BorderProps &
  system.ShadowProps &
  system.LayoutProps &
  system.FlexboxProps;

export const box = system.compose(
  system.gridArea,
  system.position,
  system.space,
  system.background,
  system.typography,
  system.color,
  system.border,
  system.shadow,
  system.layout,
  system.flexbox
);

const Box = styled('div')<BoxProps>(
  {
    boxSizing: 'border-box',
  },
  box
);

export { Box };
