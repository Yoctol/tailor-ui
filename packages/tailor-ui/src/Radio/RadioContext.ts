import { createContext } from 'react';

export type Direction = 'horizontal' | 'vertical';

const RadioContext = createContext<{
  _onChange?: (value: string) => void;
  _isChecked?: (value: string) => boolean;
  direction: Direction;
}>({
  direction: 'horizontal',
});

export { RadioContext };
