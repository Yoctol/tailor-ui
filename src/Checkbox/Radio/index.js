import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { themeGet } from 'styled-system';
import { ifProp, switchProp } from 'styled-tools';

import theme from '../../theme';

const getCheckmarkSpace = size => css`
  width: calc(${themeGet(size)} * 1.5);
  height: calc(${themeGet(size)} * 1.5);
`;

const Checkmark = styled.span`
  height: 0;
  margin-right: 0.25rem;
  border: 1px solid
    ${ifProp('disabled', themeGet('colors.border'), themeGet('colors.primary'))};
  border-radius: 50%;
  background-color: #fff;
  cursor: ${ifProp('disabled', 'default', 'pointer')};

  &::after {
    content: '';
    display: none;
    position: absolute;
    border-radius: 50%;
    background: ${themeGet('colors.primary')};
    transform: translate(-50%, -50%);

    ${switchProp('size', {
      sm: getCheckmarkSpace('space.paddingXSm'),
      m: getCheckmarkSpace('space.paddingX'),
      lg: getCheckmarkSpace('space.paddingXLg'),
    })};
  }

  ${switchProp('size', {
    sm: css`
      padding: ${themeGet('space.paddingXSm')} ${themeGet('space.paddingXSm')};
    `,
    m: css`
      padding: ${themeGet('space.paddingX')} ${themeGet('space.paddingX')};
    `,
    lg: css`
      padding: ${themeGet('space.paddingXLg')} ${themeGet('space.paddingXLg')};
    `,
  })};
`;

Checkmark.propTypes = {
  size: PropTypes.oneOf(['sm', 'm', 'lg']).isRequired,
};

Checkmark.defaultProps = {
  theme,
};

const Container = styled.div`
  display: inline-flex;
  align-items: center;
  user-select: none;

  input {
    position: absolute;
    opacity: 0;

    &:checked {
      /* stylelint-disable-next-line */
      & ~ ${Checkmark}::after {
        display: block;
      }
    }
  }
`;

const Radio = ({ disabled, size, ...props }) => (
  <Container>
    <input type="radio" disabled={disabled} {...props} />
    <Checkmark size={size} disabled={disabled} />
  </Container>
);

Radio.propTypes = {
  disabled: PropTypes.bool,
  name: PropTypes.string,
  size: PropTypes.oneOf(['sm', 'm', 'lg']),
};

Radio.defaultProps = {
  disabled: false,
  name: 'radio',
  size: 'm',
};

export default Radio;
