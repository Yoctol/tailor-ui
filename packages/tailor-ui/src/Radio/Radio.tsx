import React, {
  ChangeEventHandler,
  FocusEventHandler,
  MouseEventHandler,
  ReactNode,
  forwardRef,
  useContext,
} from 'react';

import { Box } from '../Layout';
import { useFormField } from '../FormField';

import { RadioContext } from './RadioContext';
import { RadioInner, RadioLabel, RadioWrapper, StyledRadio } from './styles';

export interface RadioProps {
  children?: ReactNode;
  checked?: boolean;
  defaultChecked?: boolean;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  value?: string;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onMouseEnter?: MouseEventHandler<HTMLLabelElement>;
  onMouseLeave?: MouseEventHandler<HTMLLabelElement>;
}

const Radio = forwardRef<HTMLLabelElement, RadioProps>(function Radio(
  {
    children,
    disabled = false,
    checked,
    defaultChecked,
    onChange,
    value,
    onMouseEnter,
    onMouseLeave,
    ...props
  },
  ref
) {
  const { handleChange, isChecked, direction } = useContext(RadioContext);
  const [, , setValue] = useFormField({
    value: checked,
    defaultValue: defaultChecked,
  });

  const inGroup = Boolean(handleChange);
  const boxChecked = isChecked && value ? isChecked(value) : checked;

  return (
    <RadioLabel
      ref={ref}
      disabled={disabled}
      direction={direction}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      <RadioWrapper>
        <StyledRadio
          disabled={disabled}
          checked={boxChecked}
          defaultChecked={defaultChecked}
          onChange={event => {
            if (onChange) {
              onChange(event);
            }
            if (handleChange && value) {
              handleChange(value);
            }
            if (!inGroup) {
              setValue(event.target.checked);
            }
          }}
          {...props}
        />
        <RadioInner />
      </RadioWrapper>
      <Box px="2">{children}</Box>
    </RadioLabel>
  );
});

export { Radio };
