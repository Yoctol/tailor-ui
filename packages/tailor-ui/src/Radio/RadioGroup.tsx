import React, { FunctionComponent, useState } from 'react';
import styled from 'styled-components';

import Radio from './Radio';
import RadioContext, { Direction } from './RadioContext';

const RadioGroupFlex = styled.div<{ direction: Direction }>`
  display: ${p => (p.direction === 'horizontal' ? 'flex' : 'inline-flex')};
  flex-direction: ${p => (p.direction === 'horizontal' ? 'row' : 'column')};
`;

export interface RadioGroupProps {
  /**
   * To set the initial value
   */
  defaultValue?: string;
  /**
   * to specify the direction of the step bar, `horizontal` and `vertical` are currently supported
   */
  direction?: Direction;
  /**
   * Specifies options
   */
  options?: {
    label: string;
    value: string;
    disabled?: boolean;
  }[];
  /**
   * Used for setting the currently selected value
   */
  value?: string;
  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (value: string) => void;
}

const RadioGroup: FunctionComponent<RadioGroupProps> = ({
  value: controlledValue,
  defaultValue,
  options = null,
  onChange,
  direction = 'horizontal' as Direction,
  children,
  ...otherProps
}) => {
  const [uncontrolledValue, setUncontrolledValue] = useState(defaultValue);

  return (
    <RadioContext.Provider
      value={{
        direction,
        _onChange: _value => {
          if (uncontrolledValue) {
            setUncontrolledValue(_value);
          }

          if (onChange) {
            onChange(_value);
          }
        },
        _isChecked: _value =>
          controlledValue
            ? controlledValue === _value
            : uncontrolledValue === _value,
      }}
    >
      <RadioGroupFlex direction={direction}>
        {options
          ? options.map(({ label, value: optionValue, disabled = false }) => (
              <Radio
                key={label}
                value={optionValue}
                disabled={disabled}
                {...otherProps}
              >
                {label}
              </Radio>
            ))
          : children}
      </RadioGroupFlex>
    </RadioContext.Provider>
  );
};

export default RadioGroup;
