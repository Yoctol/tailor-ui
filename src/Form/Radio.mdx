---
name: Radio
menu: Data Entry
route: /data-entry/radio
---

import { PropsTable, Playground } from 'docz';
import { Toggle, Value } from 'react-powerplug';
import FormField from './FormField';
import Label from './Label';
import Radio from '../Radio';
import RadioField, { RadioFieldGroup } from './RadioField';


# Radio

## Properties

### Radio

<PropsTable of={Radio} />

### RadioField

<PropsTable of={RadioField} />

### RadioField.Group

<PropsTable of={RadioFieldGroup} />

## Usage

```js
import { FormField, Label, Radio, RadioField } from '@yoctol/ui';
```

### Radio

#### Basic

<Playground>
  <FormField>
    <Radio name="radio-basic" id="radio-basic-a" defaultChecked />
    <Label htmlFor="radio-basic-a">Radio A</Label>
  </FormField>
  <FormField>
    <Radio name="radio-basic" id="radio-basic-b" />
    <Label htmlFor="radio-basic-b">Radio B</Label>
  </FormField>
</Playground>

#### With disabled

<Playground>
  <FormField>
    <Radio
      name="radio-disabled"
      id="radio-disable-a"
      checked
      disabled
    />
    <Label htmlFor="radio-disable-a">Checked Disabled</Label>
  </FormField>
  <FormField>
    <Radio name="radio-disabled" id="radio-disable-b" disabled />
    <Label htmlFor="radio-disable-b">Disabled</Label>
  </FormField>
</Playground>

#### Custom color

<Playground>
  <FormField>
    <Radio
      name="customized"
      id="custom-a"
      bg="#63bf2d"
      borderColor="#423b63"
      defaultChecked
    />
    <Label htmlFor="custom-a">Customized Color Radio A</Label>
  </FormField>
  <FormField>
    <Radio
      name="customized"
      id="custom-b"
      bg="#63bf2d"
      borderColor="#423b63"
    />
    <Label htmlFor="custom-b">Customized Color Radio B</Label>
  </FormField>
</Playground>


### RadioField

#### Basic

<Playground>
  <Value initial="radio-1">
    {({ value, set }) => (
      <div>
        <RadioField checked={value === 'radio-1'} onChange={() => set('radio-1')}>
          Radio
        </RadioField>
        <RadioField checked={value === 'radio-2'} onChange={() => set('radio-2')}>
          Radio
        </RadioField>
      </div>
    )}
  </Value>
</Playground>

### RadioField.Group

#### Basic

<Playground>
  <Value initial="radio-a">
    {({ set, value }) => (
      <RadioField.Group
        label="Radio Group"
        options={[
          { label: 'Radio A', value: 'radio-a' },
          { label: 'Radio B', value: 'radio-b' },
          { label: 'Radio C', value: 'radio-c', disabled: true },
          { label: 'Radio D', value: 'radio-d' },
        ]}
        value={value}
        onChange={set}
      />
    )}
  </Value>
</Playground>

#### With group & error

<Playground>
  <Value initial="radio-a">
    {({ set, value }) => (
      <RadioField.Group
        label="Radio Group"
        error
        message="Something went wrong!"
        options={[
          { label: 'Radio A', value: 'radio-a' },
          { label: 'Radio B', value: 'radio-b' },
          { label: 'Radio C', value: 'radio-c', disabled: true },
          { label: 'Radio D', value: 'radio-d' },
        ]}
        value={value}
        onChange={set}
      />
    )}
  </Value>
</Playground>
