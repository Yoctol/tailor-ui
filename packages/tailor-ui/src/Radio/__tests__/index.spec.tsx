import React from 'react';

import { fireEvent, render } from 'test/test-utils';

import Box from '../../Grid/Box';

import Radio from '..';

describe('Radio', () => {
  it('should render correctly', () => {
    const { container } = render(<Radio>Radio</Radio>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render disabled radio', () => {
    const { container } = render(<Radio disabled>Radio</Radio>);

    const radio = container.querySelector('input[type=radio]');

    expect(radio).toBeDisabled();
  });

  it('should render checked radio', () => {
    const { container } = render(<Radio checked>Radio</Radio>);

    const radio = container.querySelector('input[type=radio]');

    expect(radio).toHaveAttribute('checked');
  });

  it('should call onChange when radio is clicked', () => {
    const onChange = jest.fn();

    const { container } = render(
      <Radio onChange={onChange} checked={false}>
        Radio
      </Radio>
    );

    const radio = container.querySelector('input[type=radio]');

    fireEvent.click(radio);

    expect(onChange).toBeCalled();
  });
});

describe('Radio.Group', () => {
  it('should render disabled radio if option set disabled as true', () => {
    const onChange = jest.fn();

    const { container } = render(
      <Radio.Group
        direction="vertical"
        options={[
          { label: 'radio 1', value: 'radio_1', disabled: true },
          { label: 'radio 2', value: 'radio_2' },
          { label: 'radio 3', value: 'radio_3' },
        ]}
        value=""
        onChange={onChange}
      />
    );

    const radio1 = container.querySelector('input[type=radio]');

    expect(radio1).toBeDisabled();
  });

  it('should call onChange when radio is clicked', () => {
    const onChange = jest.fn();

    const { container } = render(
      <Radio.Group
        direction="vertical"
        options={[
          { label: 'radio 1', value: 'radio_1' },
          { label: 'radio 2', value: 'radio_2' },
          { label: 'radio 3', value: 'radio_3' },
        ]}
        value=""
        onChange={onChange}
      />
    );

    const radio1 = container.querySelector('input[type=radio]');

    fireEvent.click(radio1);

    expect(onChange).toBeCalledWith('radio_1');
  });

  it('should render vertical radio group', () => {
    const { container } = render(
      <Radio.Group
        direction="vertical"
        options={[
          { label: 'radio 1', value: 'radio_1' },
          { label: 'radio 2', value: 'radio_2' },
          { label: 'radio 3', value: 'radio_3' },
        ]}
        value=""
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('composition with Radio', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Radio.Group value="">
          <Box>
            <Radio value="radio_1">Radio 1</Radio>
          </Box>
          <Box>
            <Radio value="radio_2">Radio 2</Radio>
          </Box>
          <Box>
            <Radio value="radio_3">Radio 3</Radio>
          </Box>
        </Radio.Group>
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onChange when radio is clicked', () => {
      const onChange = jest.fn();

      const { container } = render(
        <Radio.Group value="" onChange={onChange}>
          <Box>
            <Radio value="radio_1">Radio 1</Radio>
          </Box>
          <Box>
            <Radio value="radio_2">Radio 2</Radio>
          </Box>
          <Box>
            <Radio value="radio_3">Radio 3</Radio>
          </Box>
        </Radio.Group>
      );

      const radio1 = container.querySelector('input[type=radio]');

      fireEvent.click(radio1);

      expect(onChange).toBeCalledWith('radio_1');
    });
  });
});
