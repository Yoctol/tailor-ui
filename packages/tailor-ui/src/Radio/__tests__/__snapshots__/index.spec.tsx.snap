// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
.c6 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c5 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 1px solid;
  border-radius: 50%;
  border-color: #bfbfbf;
  background-color: #ffffff;
  color: #373737;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c5::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  opacity: 0;
  background-color: #3c5ad0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c3 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c3:checked + .c4 {
  border-color: #3c5ad0;
}

.c3:checked + .c4::after {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  cursor: pointer;
}

.c1:hover .c4 {
  border-color: #3c5ad0;
}

.c1 + .c0 {
  margin-left: 0.5rem;
}

<label
  class="c0 c1"
  direction="horizontal"
>
  <span
    class="c2"
  >
    <input
      class="c3"
      type="radio"
    />
    <span
      class="c4 c5"
    />
  </span>
  <div
    class="c6"
    px="2"
  >
    Radio
  </div>
</label>
`;

exports[`Radio.Group composition with Radio should render correctly 1`] = `
.c1 {
  box-sizing: border-box;
}

.c8 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c7 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 1px solid;
  border-radius: 50%;
  border-color: #bfbfbf;
  background-color: #ffffff;
  color: #373737;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c7::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  opacity: 0;
  background-color: #3c5ad0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c5 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c5:checked + .c6 {
  border-color: #3c5ad0;
}

.c5:checked + .c6::after {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  cursor: pointer;
}

.c3:hover .c6 {
  border-color: #3c5ad0;
}

.c3 + .c2 {
  margin-left: 0.5rem;
}

.c9 + .c2 {
  margin-left: 0.5rem;
}

.c10 + .c2 {
  margin-top: 0.5rem;
}

.c11:hover .c6 {
  border-color: #3c5ad0;
}

.c11 + .c2 {
  margin-top: 0.5rem;
}

<div
  class="c0"
  direction="horizontal"
>
  <div
    class="c1"
  >
    <label
      class="c2 c3"
      direction="horizontal"
    >
      <span
        class="c4"
      >
        <input
          class="c5"
          type="radio"
        />
        <span
          class="c6 c7"
        />
      </span>
      <div
        class="c8"
        px="2"
      >
        Radio 1
      </div>
    </label>
  </div>
  <div
    class="c1"
  >
    <label
      class="c2 c3"
      direction="horizontal"
    >
      <span
        class="c4"
      >
        <input
          class="c5"
          type="radio"
        />
        <span
          class="c6 c7"
        />
      </span>
      <div
        class="c8"
        px="2"
      >
        Radio 2
      </div>
    </label>
  </div>
  <div
    class="c1"
  >
    <label
      class="c2 c3"
      direction="horizontal"
    >
      <span
        class="c4"
      >
        <input
          class="c5"
          type="radio"
        />
        <span
          class="c6 c7"
        />
      </span>
      <div
        class="c8"
        px="2"
      >
        Radio 3
      </div>
    </label>
  </div>
</div>
`;

exports[`Radio.Group should render vertical radio group 1`] = `
.c7 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c6 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 1px solid;
  border-radius: 50%;
  border-color: #bfbfbf;
  background-color: #ffffff;
  color: #373737;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c6::after {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  opacity: 0;
  background-color: #3c5ad0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c4 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c4:checked + .c5 {
  border-color: #3c5ad0;
}

.c4:checked + .c5::after {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c8:hover .c5 {
  border-color: #3c5ad0;
}

.c8 + .c1 {
  margin-left: 0.5rem;
}

.c9 + .c1 {
  margin-left: 0.5rem;
}

.c10 + .c1 {
  margin-top: 0.5rem;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  cursor: pointer;
}

.c2:hover .c5 {
  border-color: #3c5ad0;
}

.c2 + .c1 {
  margin-top: 0.5rem;
}

<div
  class="c0"
  direction="vertical"
>
  <label
    class="c1 c2"
    direction="vertical"
  >
    <span
      class="c3"
    >
      <input
        class="c4"
        type="radio"
      />
      <span
        class="c5 c6"
      />
    </span>
    <div
      class="c7"
      px="2"
    >
      radio 1
    </div>
  </label>
  <label
    class="c1 c2"
    direction="vertical"
  >
    <span
      class="c3"
    >
      <input
        class="c4"
        type="radio"
      />
      <span
        class="c5 c6"
      />
    </span>
    <div
      class="c7"
      px="2"
    >
      radio 2
    </div>
  </label>
  <label
    class="c1 c2"
    direction="vertical"
  >
    <span
      class="c3"
    >
      <input
        class="c4"
        type="radio"
      />
      <span
        class="c5 c6"
      />
    </span>
    <div
      class="c7"
      px="2"
    >
      radio 3
    </div>
  </label>
</div>
`;
